src/thunderbots/software/ai/navigator/obstacle/obstacle.cpp

```diff
@@ -0,0 +1,117 @@
+
+#include "obstacle.h"
+
+const Polygon& Obstacle::getBoundaryPolygon() const
+{
+    return _polygon;
+}
+
+Obstacle::Obstacle() : _polygon{Point(0, 0), Point(0, 1), Point(1, 1), Point(1, 0)} {}
+
+Obstacle::Obstacle(Polygon polygon) : _polygon(polygon) {}
+
+Obstacle Obstacle::createPlaceholderObstacle()
+{
+    return Obstacle();
+}
+
+Obstacle Obstacle::createRobotObstacle(const Robot& robot, bool enable_velocity_cushion)
+{
+    double tick_length = 1.0;
+
+    double radius_cushion = getRadiusForHexagon(ROBOT_MAX_RADIUS_METERS);
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector =
+        robot.velocity().norm(robot.velocity().len() * tick_length);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion);
+}
+
+Obstacle Obstacle::createRobotObstacleWithScalingParams(const Robot& robot,
+                                                        bool enable_velocity_cushion,
+                                                        double radius_cushion_scaling,
+                                                        double velocity_cushion_scaling,
+                                                        double tick_length)
+{
+    double radius_cushion =
+        getRadiusForHexagon(ROBOT_MAX_RADIUS_METERS * radius_cushion_scaling);
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector = robot.velocity().norm(
+        robot.velocity().len() * tick_length * velocity_cushion_scaling);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion && velocity_cushion_vector.len() > radius_cushion);
+}
+
+Obstacle Obstacle::createRobotObstacleWithBufferParams(
+    const Robot& robot, bool enable_velocity_cushion,
+    double additional_radius_cushion_buffer, double additional_velocity_cushion_buffer,
+    double tick_length)
+{
+    double radius_cushion =
+        getRadiusForHexagon(ROBOT_MAX_RADIUS_METERS + additional_radius_cushion_buffer);
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector = robot.velocity().norm(
+        robot.velocity().len() * tick_length + additional_velocity_cushion_buffer);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion && velocity_cushion_vector.len() > radius_cushion);
+}
+
+
+Obstacle Obstacle::createRobotObstacleFromPositionAndRadiusAndVelocity(
+    Point position, double radius_cushion, Vector velocity_cushion_vector,
+    bool enable_velocity_cushion)
```

Comment:
Hmm, I can't think of a situation where we would want to construct a Robot Obstacle without a Robot. Unless there is a specific use case we need this for, I'd suggest removing it.

Reply:
This is a helper that is private

Reply:
Ah, sorry, should have noticed that. My bad.